talk/objectorientation/objectsclasses.tex

```diff
@@ -108,21 +108,26 @@
 
 \begin{frame}[fragile]
   \frametitlecpp[98]{{\ttfamily this} keyword}
-  \begin{block}{}
+  \begin{block}{How to know an object's address?}
     \begin{itemize}
-    \item {\ttfamily this} is a hidden parameter to all class methods
-    \item it points to the current object
-    \item so it is of type {\ttfamily T*} in the methods of class {\ttfamily T}
+    \item Sometimes we need to know where our class data is located
+    \item For example to implement operators, see later
+    \item All class methods can use the keyword \mintinline{cpp}{this}
+      \begin{itemize}
+        \item It returns the address of the current object
+        \item Its type is \mintinline{cpp}{T*} in the methods of a struct/class {\ttfamily T}
+      \end{itemize}
     \end{itemize}
   \end{block}
   \begin{cppcode}
-    void ext_func(MyFirstClass& c) {
-      ... do something with c ...
-    }
+    struct MyStruct {
+      MyStruct * whereAreYou() {
+        return this;
+      }
+    };
 
-    int MyFirstClass::some_method(...) {
-      ext_func(*this);
-    }
+    MyStruct c;
+    c.whereAreYou(); // (MyStruct *) 0x10340e0f8
```

Comment:
I preferred the previous example. `whereAreYou` is not really a function you would write in real life. As already said, who cares where it is ? But being able to call a function that expects a reference to yourself, that's definitely an everyday need.

Reply:
OK, I created an issue for myself.